lumicks/pylake/force_calibration/calibration_item.py

```diff
@@ -130,6 +160,43 @@ def __repr__(self):
         )
         return f"{self.__class__.__name__}({properties})"
 
+    def _check_has_data(self):
+        if not self.voltage:
+            raise ValueError(
+                "This calibration item does not contain the raw data. If you still have the "
+                "timeline force data, you can de-calibrate that and perform the re-calibration"
+                "manually. See the pylake tutorial on force calibration for more information."
+            )
+
+    def plot(self):
+        self._check_has_data()
+        self.recalibrate_with().plot()
+
+    def plot_spectrum_residual(self):
+        """Plot the residuals of the fitted spectrum.
+
+        This diagnostic plot can be used to determine how well the spectrum fits the data. While
+        it cannot be used to diagnose over-fitting (being unable to reliably estimate parameters
+        due to insufficient information in the data), it can be used to diagnose under-fitting (the
+        model not fitting the data adequately).
+
+        In an ideal situation, the residual plot should show a noise band around 1 without any
+        systematic deviations.
+        """
+        self._check_has_data()
+        self.recalibrate_with().plot_spectrum_residual()
```

Comment:
why do you need an empty recalibrate here?

Reply:
Because I need to do the calibration. There is no calibration stored, only raw data, so I need to actually perform the calibration with all the raw data in order to show the spectra and fit.